src/main/java/com/example/demo/service/UserService.java

```diff
@@ -38,18 +41,23 @@ public User registerUser(RegistrationRequest request) {
     }
     var newUser =
         UserDTO.builder()
-            .firstName(request.firstName())
-            .lastName(request.lastName())
-            .email(request.email())
+            .login(request.login())
             .password(passwordEncoder.encode(request.password()))
-            .role(request.role())
             .build();
+
     return userRepository.save(mapper.userDtoToUser(newUser));
   }
 
   @Override
   public List<UserDTO> findByEmail(String email) {
-    return mapper.listOfUserToListOfUserDto(userRepository.findByEmail(email));
+    List<UserDataDTO> userData = mapper.listOfUserDataToListOfUserDataDto(userDataRepository.findByEmail(email));
+    List<User> user;
+    if (userData.isEmpty()){
+      return null;
```

Comment:
try to avoid returning nulls. It is better to return Optional or at leas empty object